src/dxil/wrapper.rs

```diff
@@ -55,10 +55,8 @@ impl DxcContainerReflection {
         Self { inner }
     }
 
-    pub fn load(&self, p_dxc_blob: &DxcBlob) {
-        unsafe {
-            self.inner.load(p_dxc_blob.inner.as_ptr());
-        }
+    pub fn load(&self, p_dxc_blob: &DxcBlob) -> HassleError {
+        unsafe { HassleError::Win32Error(self.inner.load(p_dxc_blob.inner.as_ptr())) }
```

Comment:
@DBouma This always returns an error, shouldn't this be:

```suggestion
    pub fn load(&self, p_dxc_blob: &DxcBlob) -> Result<()> {
        unsafe { check_hr_wrapped!(self.inner.load(p_dxc_blob.inner.as_ptr()), ()) }
```

@Jasper-Bekkers On this note we currently have `Result<..., HRESULT>` with `check_hr!`, and `Result<..., HassleError>` with `check_hr_wrapped!`. Should we clean this up and use the latter one consistently across all public-facing functions?

Reply:
The error contains `S_OK`, or `0x0000000` upon succes. The suggested breaks the macro.

Reply:
Returning `HassleError` makes it impossible for the caller to check whether it is `S_OK` as this "newtype"-style enum is private: `Win32Error(HRESULT),`. Besides, returning `HassleError::Win32Error` implies there's always some sort of error.

Returning `Result<(), HassleError>` instead clarifies that this function may possibly fail, but otherwise yield no result (ie. `Ok(())`). This allows using all the usual `?`, `.unwrap()` or `.expect()` operations we've all come accustomed to.

The suggested change failing is me mistyping this as pseudo-code. This should compile:

```suggestion
    pub fn load(&self, p_dxc_blob: &DxcBlob) -> Result<(), HassleError> {
        return_hr_wrapped!(unsafe { self.inner.load(p_dxc_blob.inner.as_ptr() }, ());
```

Keep in mind that I intentionally used `check_hr_wrapped` instead of `return_hr_wrapped` which is more flexible and already merged to master. I'm sorry for making that change while you were working on this as it's likely conflicting quite a bunch. But for new code it should be as simple as a search+replace and removing the trailing `;`.